### PR TITLE
Notes-duplication

### PR DESCRIPTION
**Pull request by Farajialireza82**

### CODE REVIEW

File: app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 public class AddNoteActivity extends AppCompatActivity {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java]: chera kotlin nemikoni in class ro?

File: app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java [hunk 2]
```diff
@@ -15,6 +15,9 @@
...
 
+    public static String EXTRA_NOTE = "com.example.arc_exapmle.EXTRA_NOTE";
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java]: age kotlin kardi 
baraye in moteghayer az ravesh hayi ke dishab yad gerefti estefade kon

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt [hunk 3]
```diff
@@ -88,15 +85,17 @@ class MainActivity : AppCompatActivity() {
         super.onActivityResult(requestCode, resultCode, data)
 
         if (requestCode == 1 && resultCode == Activity.RESULT_OK) {
-            var title = data?.getStringExtra(EXTRA_TITLE)
-            var description = data?.getStringExtra(EXTRA_Description)
-            var priority = data?.getIntExtra(EXTRA_PRIORITY, 1)
+            val mNote = data?.getParcelableExtra<Note>(EXTRA_NOTE)
 
+            if (mNote != null) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: afarin

File: app/src/main/java/com/example/arc_exapmle/Note.kt [hunk 4]
```diff
@@ -1,18 +1,28 @@
...
-    private var description: String?,
-    private var priority: Int?
-) {
+     var title: String?,
+     var description: String?,
+     var priority: Int?
+) : Parcelable {
 
     @PrimaryKey(autoGenerate = true)
     private var id = 0
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/Note.kt]: in chera birone?

File: app/src/main/java/com/example/arc_exapmle/Note.kt [hunk 5]
```diff
@@ -1,18 +1,28 @@
...
 
+    constructor(parcel: Parcel) : this(
+        parcel.readString(),
+        parcel.readString(),
+        parcel.readValue(Int::class.java.classLoader) as? Int
+    ) {
+        id = parcel.readInt()
+    }
+
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/Note.kt]: be ina niazi nist
shoma az @Parcelize estefade kon!
in chiza ham pak kon

File: app/src/main/java/com/example/arc_exapmle/Note.kt [hunk 6]
```diff
@@ -1,18 +1,28 @@
...
 @Entity(tableName = "note_table")
 class Note(
-    private var title: String?,
-    private var description: String?,
-    private var priority: Int?
-) {
+     var title: String?,
+     var description: String?,
+     var priority: Int?
+) : Parcelable {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/Note.kt]: ya khoda
Parcelable esh ham kardi?
yadete goftam har class faghat yekar bokone behtare?!

File: app/src/main/java/com/example/arc_exapmle/Note.kt [hunk 7]
```diff
@@ -1,18 +1,28 @@
 package com.example.arc_exapmle
 
+import android.os.Parcel
+import android.os.Parcelable
 import androidx.room.Entity
 import androidx.room.PrimaryKey
 
 @Entity(tableName = "note_table")
 class Note(
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/Note.kt]: gharar bod data class beshe ke!

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package com.example.arc_exapmle
+
+class NoteUI(noteEntity: NoteEntity?) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: chera data class nist?

Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: chera dari vorodi NoteEntity migiri?

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    internal var title: String? = null
+    internal var description: String? = null
+    internal var priority: Int? = null
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: chera internal gozashti?
chera nullable hastan?

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+    init {
+        title = noteEntity?.title
+        description = noteEntity?.description
+        priority = noteEntity?.priority
+    }
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: in karo bayad biron az in class anjam bedi
in class nabayad niazi be NoteEntity dashte bashe.
bayad faghat parameter haye vorodi sho begire.
dar zemn behtare id ham pas bedi

File: app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt [hunk 11]
```diff
@@ -8,7 +8,7 @@ import androidx.recyclerview.widget.RecyclerView
 
 class NoteAdapter : RecyclerView.Adapter<NoteAdapter.NoteHolder>() {
 
-    private var notes: List<NoteEntity>? = ArrayList()
+    private var notes: List<NoteUI>? = ArrayList()
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt]: nemifahmam chera hanoz nullable hastesh

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt [hunk 12]
```diff
@@ -50,7 +50,22 @@ class MainActivity : AppCompatActivity() {
...
-            Observer { t -> adapter.setNote(t) }
+            Observer { t: List<NoteEntity>? ->
+
+                val noteUIList: MutableList<NoteUI> = ArrayList()
+
+
+
+                for (i in t!!.indices) {
+
+                    noteUIList.add(NoteUI(t[i]))
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: behtare toye ViewModel
NoteEntity ro be NoteUI tabdil koni baad be in livedata bedi
intori faghat settesh mikoni roye adapter.

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java (hunk 1 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java]: Kotlin shod

File: app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java (hunk 2 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/AddNoteActivity.java]: Halle . az companion Object estefade kardam

File: app/src/main/java/com/example/arc_exapmle/Note.kt (hunk 4 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/Note.kt]: In class ro kolan pak kardam az khode project am . amma methl in ke az GitHub pak nemishe

File: app/src/main/java/com/example/arc_exapmle/Note.kt (hunk 5 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/Note.kt]: in ham methl ghabli pak shode classesh vali baray classe jaigozinesh in kar ro anjam medim . chashm

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt (hunk 9 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: In ro ham dorost kardam va taghiresh dadam

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt (hunk 10 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: anjam shod

File: app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt (hunk 11 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt]: in haro ham kamel hal kardam

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt (hunk 12 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: in ham belakhare hal shod .

File: app/src/main/java/com/example/arc_exapmle/Note.kt (hunk 6 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/Note.kt]: hal shod in ham

File: app/src/main/java/com/example/arc_exapmle/Note.kt (hunk 7 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/Note.kt]: in class kollan hazf shode

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt [hunk 13]
```diff
@@ -4,24 +4,18 @@ import android.os.Parcel
...
 @Entity(tableName = "note_table")
 class NoteEntity(
-     var title: String?,
-     var description: String?,
-     var priority: Int?
-) :  Parcelable {
+     var title: String,
+     var description: String,
+     var priority: Int
+) :  Serializable {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteEntity.kt]: chera intorish kardi?
man migam parcelable ro bardar 
serializable esh mikoni?

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt [hunk 14]
```diff
@@ -99,7 +109,7 @@ class MainActivity : AppCompatActivity() {
         super.onActivityResult(requestCode, resultCode, data)
 
         if (requestCode == 1 && resultCode == Activity.RESULT_OK) {
-            val mNote = data?.getParcelableExtra<NoteEntity>(AddNoteKtActivity.EXTRA_NOTE)
+            val mNote = data?.getSerializableExtra(AddNoteKtActivity.EXTRA_NOTE) as NoteEntity
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: note entity ro nafrest invar onvar 
LocalModel
ya
UI Model besaz

File: app/src/main/java/com/example/arc_exapmle/NoteRepository.kt [hunk 15]
```diff
@@ -39,43 +39,48 @@ class NoteRepository(application: Application) {
...
     }
 
-    fun getAllNotes(): LiveData<List<NoteEntity>?> {
+    fun getAllNotes(): LiveData<List<NoteEntity>> {
+
         return allNotes
     }
 
-    private class InsertNoteAsyncTask(noteDaoNew: NoteDao) : AsyncTask<NoteEntity, Unit, Unit>() {
+    private class InsertNoteAsyncTask(noteDaoNew: NoteDao) : AsyncTask<NoteUI, Unit, Unit>() {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteRepository.kt:47]: NoteUi ro behtare to ViewModel tabdil be NoteEntity koni
baad befresti baraye repo

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt [hunk 16]
```diff
@@ -1,16 +1,3 @@
...
-        title = noteEntity?.title
-        description = noteEntity?.description
-        priority = noteEntity?.priority
-    }
-
-
-
-}
\ No newline at end of file
+class NoteUI( var id: Int, var title: String, var description: String, var priority: Int)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: chera data class estefade nakardi?

Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteUI.kt]: Anjam shod . Parcelizesh ham kardam

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt (hunk 14 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: Anjamesh dadam

File: app/src/main/java/com/example/arc_exapmle/NoteRepository.kt (hunk 15 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteRepository.kt:47]: manam hamin kar ro anjam dadam diige 
albate hala ye taghirati behesh dadam 
bebinin ishalla hal shode bashe in masale ham

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt [hunk 17]
```diff
@@ -1,3 +1,8 @@
 package com.example.arc_exapmle
 
-class NoteUI( var id: Int, var title: String, var description: String, var priority: Int)
\ No newline at end of file
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+data class NoteUI(var id: Int, var title: String , var description: String , var priority: Int) :
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteUI.kt:7]: chera var hastan?
chera az val estefade nakardi?

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt [hunk 18]
```diff
@@ -8,21 +8,21 @@ import java.io.Serializable
...
-     var description: String,
-     var priority: Int
-) :  Serializable {
+    var title: String,
+    var description: String,
+    var priority: Int
+) {
 
     @PrimaryKey(autoGenerate = true)
     private var id = 0
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteEntity.kt]: ino chera nemibari to constructor?

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt [hunk 19]
```diff
@@ -8,21 +8,21 @@ import java.io.Serializable
 
 @Entity(tableName = "note_table")
 class NoteEntity(
-     var title: String,
-     var description: String,
-     var priority: Int
-) :  Serializable {
+    var title: String,
+    var description: String,
+    var priority: Int
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteEntity.kt:11]: chera val nistan ina?

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt [hunk 20]
```diff
@@ -8,21 +8,21 @@ import java.io.Serializable
...
+) {
 
     @PrimaryKey(autoGenerate = true)
     private var id = 0
 
 
-    fun setId(id : Int){
+    fun setId(id: Int) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteEntity.kt]: niazi ne getter ya setter nist 
khode kotlin dorost mikone hame chio

File: app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt [hunk 21]
```diff
@@ -24,11 +24,11 @@ class NoteAdapter : RecyclerView.Adapter<NoteAdapter.NoteHolder>() {
 
         val currentNote: Note? = notes?.get(position)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/NoteAdapter.kt]: chera notes hanoz nullable hastesh?

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt [hunk 22]
```diff
@@ -109,17 +120,16 @@ class MainActivity : AppCompatActivity() {
         super.onActivityResult(requestCode, resultCode, data)
 
         if (requestCode == 1 && resultCode == Activity.RESULT_OK) {
-            val mNote = data?.getSerializableExtra(AddNoteKtActivity.EXTRA_NOTE) as NoteEntity
 
-            if (mNote != null) {
-                noteViewModel.insert(mNote)
-                Toast.makeText(this, "NoteSaved", Toast.LENGTH_SHORT).show()
+            val mNote = data?.getParcelableExtra(AddNoteKtActivity.EXTRA_NOTE) as NoteUI
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: afarin

File: app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt [hunk 23]
```diff
@@ -61,9 +62,10 @@ class AddNoteKtActivity : AppCompatActivity() {
 
         val note = NoteEntity(title, description, priority)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt]: chera inja NoteEntity misazi?

File: app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt [hunk 24]
```diff
@@ -61,9 +62,10 @@ class AddNoteKtActivity : AppCompatActivity() {
...
 
+
         val data = Intent()
 
-        data.putExtra(EXTRA_NOTE, note)
+        data.putExtra(EXTRA_NOTE, NoteUI(note.getId(), note.title, note.description, note.priority))
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt]: va chera in payin tabdilesh mikoni be ui?
chera mostaghim nemizari to UI?

File: app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt [hunk 25]
```diff
@@ -61,9 +62,10 @@ class AddNoteKtActivity : AppCompatActivity() {
...
         setResult(Activity.RESULT_OK, data)
```
Comment (helloseyedjafari) [inline app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt]: mitoni bejaye activityResult
Note ro bezari to DB va DB khodesh baghie safahat ro khabar mikone

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/arc_exapmle/NoteUI.kt (hunk 17 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteUI.kt:7]: too net khondsm ke baraye inke @Parcelize kar kone bayad in maghadir "var" bashan

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt (hunk 20 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteEntity.kt]: in har vaght getter nadasht error midad . nemidonam chera . yeja khondam buge

File: app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt (hunk 23 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt]: baraie inke khode Room barash yek ID besaze

File: app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt (hunk 24 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/AddNoteKtActivity.kt]: bekhater hamon masaleye comment ghabli

File: app/src/main/java/com/example/arc_exapmle/MainActivity.kt (hunk 22 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/MainActivity.kt]: :)

File: app/src/main/java/com/example/arc_exapmle/NoteEntity.kt (hunk 18 above)
Comment (Farajialireza82) [reply app/src/main/java/com/example/arc_exapmle/NoteEntity.kt]: hal shod :)

**Pull request merged**